content/blog/registry-writing-i2c-component/index.md

```diff
@@ -0,0 +1,742 @@
+
+The new features for this new driver includes:
+
+- New initialization mode.
+- Thread safety.
+- The APIs were simplified and some were removed.
+
+Please not if you are still using the legacy driver, update to the new driver since the legacy will be removed in the future.
+
+### Creating the component
+
+The Component Manager [documentation](https://docs.espressif.com/projects/idf-component-manager/en/latest/) is very comprehensive and you will find all the information you need there, including the [simple component](https://docs.espressif.com/projects/idf-component-manager/en/latest/guides/packaging_components.html#a-simple-esp-idf-component) structure.
+
+To create the component, we will go through some steps:
+
+1. Create the new component using the command line.
+2. Create the component manifest file and set with the component information.
+3. Set the license.
+4. Write the README.md file.
+5. Write the code for the sensor driver.
+6. Write the example.
+
+We assume that you already have the ESP-IDF installed on your system. If you don't, please check our article about installing the ESP-IDF.
+
+On the Espressif DevCon23, Ivan Grokhotkov, gave a talk titled: Developing, Publishing, and Maintaining Components for ESP-IDF. You can watch this talk as an additional material for your studies.
+
+{{< youtube D86gQ4knUnc >}}
+
+#### Create the new component
+
+The process will be done by the CLI (Command-line Interface) tool `idf.py`.
+
+```bash
+idf.py create-component shtc3
+```
+
+After that, the new folder should contain the following structure:
+
+```text
+.
+└── shtc3
+    ├── CMakeLists.txt
+    ├── include
+    │   └── shtc3.h
+    └── shtc3.c
+```
+
+#### Create the Manifest file
+
+The manifest file is mandatory and this is the file that the Registry will recognize it as a component. The bare-minimum for this file is described in the documentation, but we will fill with the most common important fields.
+
+Create the `idf_component.yml` in the component root directory with this content. You can change it if you want.
+
+```yaml
+version: 1.0.0
+targets:
+  - esp32
+  - esp32s2
+  - esp32s3
+  - esp32c3
+  - esp32c2
+  - esp32c6
+  - esp32h2
+  - esp32p4
+description: SHTC3 Temperature and humidity sensor driver for ESP-IDF
+url:
+repository:
+issues:
+maintainers:
+  - "Author Name <author@email.com>"
+tags:
+  - shtc3
+  - sensor
+  - temperature
+  - humidity
+  - i2c
+  - driver
+dependencies:
+  idf : ">=5.3"
+```
+
+- **version**: This contains the component version following the version format `major.minor.patch` (0.0.0).
+- **targets**: This is the list of the supported targets.
+- **description**: Brief description about the component.
+- **url**: URL to the component page or company page.
+- **repository**: URL to the GitHub component repository ending with `.git`.

```

Comment:
There are two options, using `https://` or `git@`. We should recommend the first option for public archives, since it does not requires authentication on GitHub.